Shorten identifier for two reasons (#3685)
modelica/ModelicaSpecification: chapters/functions.tex

  \end{nonnormative}
\item

>   The

Reviewer: Added back on the correct line in the suggestion below:
```suggestion
```

>   {\lstinline!annotation(IncludeDirectory="modelica:/PackageName/Resources/Include")!}\annotationindex{IncludeDirectory}, used to specify a location for header files.
>   The preceding one is the default and need not be specified; but another location could be specified by using an URI name for the include directory, see \cref{external-resources}.

Reviewer: I realize that something important is being lost regarding the default; it needs to be clear that the default is the name of the current library's top-level package, and `PackageName` isn't making this clear.

Something like the following would preserve the meaning and still drastically shorten the `\lstinline` arguments:
```suggestion
  {\lstinline!annotation(IncludeDirectory=$\mathit{uri}$)!}\annotationindex{IncludeDirectory}, used to specify a URI location (\cref{external-resources}) for header files.
  The default $uri$ is \lstinline!"modelica:/ModelicaLibraryName/Resources/Include"! and need not be specified.
```

Author: There is an explanation for PackageName (changed from the previous name in this PR) after the list.

Obviously we can improve that, but to me that would be a separate PR, as this was just intended to have shorter names to fix two issues.

Author: Separate PR if needed.

The directories may use symbolic links or use a text-file as described below: e.g., a text-file \filename{vs2008} containing the text \emph{../win32/vs2005} (or \emph{vs2005}) suggesting that it is compatible with vs2005.
\end{nonnormative}

> The {\lstinline!PackageName!} used for {\lstinline!IncludeDirectory!}, {\lstinline!LibraryDirectory!}, and {\lstinline!SourceDirectory!} indicates the top-level class where the annotation is found in the Modelica source code.

Author: As indicated I think that's a separate PR.

> The {\lstinline!PackageName!} used for {\lstinline!IncludeDirectory!}, {\lstinline!LibraryDirectory!}, and {\lstinline!SourceDirectory!} indicates the top-level class where the annotation is found in the Modelica source code.

Reviewer: The reason why I don't think it should be a separate PR is that while this PR is currently reduces very long lines – which looks nice – it is making it harder to see that the default location is tied to the top-level package.  This paragraph is too far away from where it applies to cover up for the less informative `PackageName` compared to the old `ModelicaLibraryName`.

> The {\lstinline!PackageName!} used for {\lstinline!IncludeDirectory!}, {\lstinline!LibraryDirectory!}, and {\lstinline!SourceDirectory!} indicates the top-level class where the annotation is found in the Modelica source code.

Author: After more thinking I believe this should be `TopPackage` or even better  `$\mathit{TopPackage}$`, not `PackageName`.

> The {\lstinline!PackageName!} used for {\lstinline!IncludeDirectory!}, {\lstinline!LibraryDirectory!}, and {\lstinline!SourceDirectory!} indicates the top-level class where the annotation is found in the Modelica source code.

Author: Except the better variant still triggered the bug due to its length.
